detection_operator_runner.py

```diff
@@ -157,11 +207,83 @@ def process_images(simulator_image):
+                    msg = erdos.Message(
+                        timestamp=timestamp,
+                        data=DepthFrame.from_simulator_frame(
+                            simulator_image,
+                            depth_camera_setup,
+                            save_original_frame=FLAGS.visualize_depth_camera))
+                    depth_camera_ingest_stream.send(msg)
+
+        def process_seg_images(simulator_image):
+            """Invoked when a segmented image is received from the simulator."""
+            game_time = int(simulator_image.timestamp * 1000)
+            timestamp = erdos.Timestamp(coordinates=[game_time])
+            watermark_msg = erdos.WatermarkMessage(timestamp)
+
+            # Ensure that the code executes serially
+            with _lock:
+                msg = None
+                if depth_camera_setup.camera_type == 'sensor.camera.semantic_segmentation':
+                    msg = erdos.Message(
+                        timestamp=timestamp,
+                        data=SegmentedFrame.from_simulator_image(
+                            simulator_image, seg_camera_setup))
+                    seg_camera_ingest_stream.send(msg)
+
+        def process_left_images(simulator_image):
+            """Invoked when an rgb image is received from the simulator."""
+            game_time = int(simulator_image.timestamp * 1000)
+            timestamp = erdos.Timestamp(coordinates=[game_time])
+            watermark_msg = erdos.WatermarkMessage(timestamp)
+
+            # Ensure that the code executes serially
+            with _lock:
+                msg = None
+                if rgb_camera_setup.camera_type == 'sensor.camera.rgb':
+                    msg = erdos.Message(timestamp=timestamp,
+                                        data=CameraFrame.from_simulator_frame(
+                                            simulator_image,
+                                            left_camera_setup))
+                    left_camera_ingest_stream.send(msg)
+                    left_camera_ingest_stream.send(watermark_msg)
+
+        def process_right_images(simulator_image):
+            """Invoked when an rgb image is received from the simulator."""
+            game_time = int(simulator_image.timestamp * 1000)
+            timestamp = erdos.Timestamp(coordinates=[game_time])
+            watermark_msg = erdos.WatermarkMessage(timestamp)
+
+            # Ensure that the code executes serially
+            with _lock:
+                msg = None
+                if rgb_camera_setup.camera_type == 'sensor.camera.rgb':
+                    msg = erdos.Message(timestamp=timestamp,
+                                        data=CameraFrame.from_simulator_frame(
+                                            simulator_image,
+                                            right_camera_setup))
+                    right_camera_ingest_stream.send(msg)
+                    right_camera_ingest_stream.send(watermark_msg)
```

Comment:
Is it possible to simplify these by creating an `add_carla_callback` function that takes the stream, the config, and the CARLA sensor and calls `sensor.listen(...)` on the proper callback? I'm thinking something like:

```python
def add_carla_callback(carla_sensor, setup, stream):
   def callback(simulator_data):
      if getattr(setup, "camera_type") == "sensor.camera.rgb":
         ...
      elif getattr(setup, "camera_type") == "sensor.camera.depth":
         ....
   sensor.listen(callback)
```
Basically, I'm hoping to reduce the code duplication, and to make it easier to add additional sensors.